plugins/speculation-rules/tests/test-speculation-rules-settings.php

```diff
@@ -122,4 +122,152 @@ public function test_plsr_add_settings_action_link(): void {
 			plsr_add_settings_action_link( $default_action_links )
 		);
 	}
+
+	/**
+	 * @covers ::plsr_get_mode_labels
+	 * @covers ::plsr_get_eagerness_labels
+	 * @covers ::plsr_get_setting_default
+	 * @covers ::plsr_register_setting
+	 */
+	public function test_register_settings(): void {
+		plsr_register_setting();
+		$settings = plsr_get_setting_default();
+		$this->assertArrayHasKey( 'mode', $settings );
```

Comment:
If we check this, we should probably also check for the `eagerness` key.
```suggestion
		$this->assertArrayHasKey( 'mode', $settings );
		$this->assertArrayHasKey( 'eagerness', $settings );

```